soundcloud-api/src/main/java/com/jlubecki/soundcloud/webapi/android/models/Activity.java

```diff
@@ -0,0 +1,58 @@
+package com.jlubecki.soundcloud.webapi.android.models;
+
+/**
+ * Created by takahawk on 7/10/16.
+ */
+public class Activity {
+
+    public ActivityOrigin origin;
```

Comment:
Your `ActivityOrigin` is effectively the same as a `Track`. It would make more sense to delete the `ActivityOrigin` class and just change this line to `public Track origin;`. I'm thinking about adding some custom deserialization to handle cases where the origin isn't a track object, but I'm not sure it makes sense to make a class that is represented by an existing model.